test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,386 @@
+        dataset.data(dataWithNaN);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is NaN");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned  when one of the y values is NaN");
+
+        dataWithNaN = data.slice();
+        dataWithNaN[2] = { x: NaN, y: 0.4 };
+        dataset.data(dataWithNaN);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the x values is NaN");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the x values is NaN");
+
+        let dataWithUndefined = data.slice();
+        dataWithUndefined[2] = { x: 0.4, y: undefined };
+        dataset.data(dataWithUndefined);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is undefined");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the y values is undefined");
+
+        dataWithUndefined = data.slice();
+        dataWithUndefined[2] = { x: undefined, y: 0.4 };
+        dataset.data(dataWithUndefined);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the x values is undefined");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the x values is undefined");
+
+        let dataWithInfinity = data.slice();
+        dataWithInfinity[2] = { x: 0.4, y: Infinity };
+        dataset.data(dataWithInfinity);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is Infinity");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the y values is Infinity");
+
+        dataWithInfinity = data.slice();
+        dataWithInfinity[2] = { x: -Infinity, y: 0.4 };
+        dataset.data(dataWithInfinity);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the x values is -Infinity");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the x values is -Infinity");
+
+        let dataWithString = data.slice();
+        dataWithString[2] = { x: 0.4, y: "12" };
+        dataset.data(dataWithString);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is \"12\"");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the y values is \"12\"");
+
+        dataWithString = data.slice();
+        dataWithString[2] = { x: "abc", y: 0.4 };
+        dataset.data(dataWithString);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the x values is \"abc\"");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned when one of the x values is \"abc\"");
+        svg.remove();
+      });
+
+      it("shows warning when deprecated method is called", () => {
```

Comment:
I suggest delete this test. We are testing a deprecated private method. This cannot be right.